iroha_p2p/src/network.rs

```diff
@@ -0,0 +1,123 @@
+use std::collections::HashMap;
+use std::io::Error;
+
+use async_stream::stream;
+use futures::Stream;
+use tokio::net::{TcpListener, TcpStream};
+
+use iroha_actor::{Actor, Addr, Context, Handler, AlwaysAddr};
+use iroha_actor::broker::Broker;
+use iroha_crypto::{PublicKey};
+use iroha_logger::warn;
+
+use crate::peer::{Peer, State};
+use iroha_data_model::peer::prelude::PeerId;
+use ursa::encryption::symm::Encryptor;
+
+pub struct Network<E: Encryptor + Send + 'static> {
+    peers: HashMap<PeerId, Addr<Peer<E>>>,
+    listener: Option<TcpListener>,
+    addr: Option<Addr<Self>>,
+    broker: Broker,
+    //handler: AlwaysAddr<A>
+}
+
+impl<E: Encryptor + Send + 'static> Network<E> {
+    pub fn new(broker: Broker, /*handler: AlwaysAddr<A>*/) -> Self {
+        Self { peers: HashMap::new(), listener: None, addr: None, broker, /*handler*/ }
+    }
+
+    fn listener_stream(&mut self) -> impl Stream<Item = NewPeerResult> + Send + 'static {
+        let listener = self.listener.take().unwrap();
+        stream! {
+            loop {
+                match listener.accept().await {
+                    Ok((stream, addr)) => {
+                        let id = PeerId { address: addr.to_string(), public_key: PublicKey::default() };
+                        let new_peer: NewPeerResult = NewPeerResult::Ok { stream, id };
+                        yield new_peer;
+                    },
+                    Err(error) => {
+                        yield NewPeerResult::Err { error };
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    async fn connect_peer(&mut self, id: PeerId) {
+        let my_addr = self.addr.clone().expect("Error connecting peer from unstarted Network!");
+        let addr = Peer::new(id.clone(), None, State::Connecting, my_addr).start().await;
+        self.peers.insert(id, addr);
+    }
```

Comment:
This function is not used anywhere... 😨

Reply:
Sure, this function will be called from `Torii` or another user of Network.

Reply:
Network is an actor... You can send message to it and network can handle it. So all api should be in terms of messages

Reply:
Okay, implemented it!